critiquebrainz/db/users.py

```diff
@@ -70,3 +73,543 @@ def get_many_by_mb_username(usernames):
+            """), {
+                "revision_id": last_revision['id'],
+                "user_id": user_id
+            })
+        count = result.fetchone()[0]
+    return count > 0
+
+
+def karma(user_id):
+    """Get the karma of a user
+
+    Args:
+        user_id(uuid): ID of the user
+
+    Returns:
+        karma(int): the karma of the user
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT review.user_id, vote
+              FROM ((vote
+         LEFT JOIN revision
+                ON revision.id = revision_id)
+         LEFT JOIN review
+                ON review.id = review_id)
+         LEFT JOIN "user"
+                ON "user".id = review.user_id
+             WHERE "user".id = :user_id
+        """), {
+            "user_id": user_id
+        })
+
+        rows = result.fetchall()
+        _karma = 0
+        for row in rows:
+            if row.vote == True:
+                _karma += 1
+            else:
+                _karma -= 1
+    return _karma
+
+
+def reviews(user_id):
+    """Get list of reviews written by a user
+
+    Args:
+        user_id(uuid): ID of the user
+
+    Returns:
+        List of reviews written by the user with the structure:
+        {
+            "id": (uuid)
+            "entity_id": (uuid),
+            "entity_type": event or place or release_group,
+            "user_id": (uuid),
+            "edits": (int),
+            "is_draft": (bool),
+            "license_id": (str),
+            "language": (str),
+            "source": (str),
+            "source_url" (str)
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT id, entity_id, entity_type, user_id, edits,
+                   is_draft, license_id, language, source, source_url
+              FROM review
+             WHERE user_id = :user_id
+        """), {
+            "user_id": user_id
+        })
+
+        rows = result.fetchall()
+        rows = [dict(row) for row in rows]
+
+    return rows
+
+
+def get_votes(user_id, date='1-1-1970'):
+    """Get votes by a user from a specified time
+
+    Args:
+        user_id(uuid): ID of the user
+        date(datetime): Date from which votes submitted(default: UTC)
+
+    Returns:
+        List of votes submitted by the user from the time specified
+        {
+            "vote": (bool),
+            "rated_at": (datetime)
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT vote, rated_at
+              FROM vote
+             WHERE user_id = :user_id
+               AND rated_at >= :date
+        """), {
+            "user_id": user_id,
+            "date": date
+        })
+
+        rows = result.fetchall()
+        rows = [dict(row) for row in rows]
+
+    return rows
+
+
+def get_reviews(user_id, date='1-1-1970'):
```

Comment:
In `db/users.py`, `get_reviews` and  `get_votes` were used  for `reviews_since` and `votes_since` in `User` class (to fetch reviews and votes from a certain date). To fetch all reviews or votes, I set`date` to start of Unix time. I changed the function name `get_reviews_since` and `get_vote_since` to increase understanding. Should it be done in some other preferred way?